### PR TITLE
removed use of getlantern proxy module closes #3876

### LINKED ISSUE #3876: getlantern/proxy package not used in practice
chained/dialer.go implements the proxy.Dailer interface, but nothing external references chained that way.

### PR DESCRIPTION
**Pull request by myleshorton**

### CODE REVIEW

**Comment by myleshorton** [comment]
You mind looking at this one @oxtoacart?

File: src/github.com/getlantern/chained/dialer.go [hunk 1]
```diff
@@ -106,7 +100,7 @@ func checkCONNECTResponse(r *bufio.Reader, req *http.Request) error {
 
 func sameStatusCodeClass(statusCode1 int, statusCode2 int) bool {
 	// HTTP response status code "classes" come in ranges of 100.
-	var classRange int = 100
+	const classRange = 100
```
Comment (myleshorton) [inline src/github.com/getlantern/chained/dialer.go]: The linter flagged this...

File: src/github.com/getlantern/flashlight/client/chained.go [hunk 2]
```diff
@@ -19,10 +19,10 @@ import (
 // to eliminate "too many open files" error.
 var idleTimeout = 1 * time.Hour
 
-// If specified, all proxying will go through this address
+// ForceChainedProxyAddr - If specified, all proxying will go through this address
 var ForceChainedProxyAddr string
 
-// If specified, auth token will be forced to this
+// ForceAuthToken - If specified, auth token will be forced to this
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/client/chained.go]: Linter...

File: src/github.com/getlantern/flashlight/client/chained.go [hunk 3]
```diff
@@ -19,10 +19,10 @@ import (
 // to eliminate "too many open files" error.
 var idleTimeout = 1 * time.Hour
 
-// If specified, all proxying will go through this address
+// ForceChainedProxyAddr - If specified, all proxying will go through this address
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/client/chained.go]: Linter...

File: src/github.com/getlantern/flashlight/client/chained.go [hunk 4]
```diff
@@ -86,8 +86,8 @@ func (s *ChainedServerInfo) Dialer(deviceID string) (*balancer.Dialer, error) {
 				return nil, err
 			}
 			if !forceProxy && !conn.ConnectionState().PeerCertificates[0].Equal(x509cert) {
-				if err := conn.Close(); err != nil {
-					log.Debugf("Error closing chained server connection: %s", err)
+				if errr := conn.Close(); err != nil {
+					log.Debugf("Error closing chained server connection: %s", errr)
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/client/chained.go]: Linter again

File: src/github.com/getlantern/flashlight/client/chained.go [hunk 5]
```diff
@@ -124,7 +124,7 @@ func (s *ChainedServerInfo) Dialer(deviceID string) (*balancer.Dialer, error) {
 		Label:   label,
 		Trusted: s.Trusted,
 		DialFN: func(network, addr string) (net.Conn, error) {
-			conn, err := d.Dial(network, addr)
+			conn, err := d(network, addr)
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/client/chained.go]: Primary change along with the switch to just return the function in chained/dialer.go

File: src/github.com/getlantern/chained/dialer.go [hunk 6]
```diff
@@ -24,20 +22,21 @@ type Config struct {
 	Label string
 }
 
-// dialer is an implementation of proxy.Dialer that proxies traffic via an
-// upstream server proxy.  Its Dial function uses DialServer to dial the server
+// dialer is an implementation of net.Dial that proxies traffic via an
```
Comment (oxtoacart) [inline src/github.com/getlantern/chained/dialer.go]: net.Dial is a function, so technically dialer is not an implementation of it.

**Pull request merged**